tools/c7n_azure/c7n_azure/resources/generic_arm_resource.py

```diff
@@ -12,10 +12,34 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from c7n_azure.constants import RESOURCE_GROUPS_TYPE
 from c7n_azure.provider import resources
+from c7n_azure.query import DescribeSource, ResourceQuery
 from c7n_azure.resources.arm import ArmResourceManager, arm_resource_types
-from c7n_azure.utils import ResourceIdParser
+
 from c7n.filters.core import Filter, type_schema
+from c7n.query import sources
+from c7n_azure.utils import ResourceIdParser, is_resource_group_id
+
+
+class GenericArmResourceQuery(ResourceQuery):
+
+    def filter(self, resource_manager, **params):
```

Comment:
I'm a little confused at how this method is filtering. Isn't this just fetching all the resources and resource groups?

Reply:
Never mind saw that there is a comment on the c7n core ResourceQuery filter method that it just queries for resources! In what scenarios are we supposed to use ResourceQuery `filter` to fetch the resources over the QueryResourceManager `get_resources` method?

Reply:
I think it was real `filter` a long time ago, but now it is just query :)

`get_resources` returns only resources with a specific ids, so we use it only for events now (and probably for related filters)

Reply:
Oh ok that makes sense - seems their names should have been swapped lol.